pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -345,8 +344,8 @@ func (a *Actuator) Update(ctx context.Context, cluster *clusterv1.Cluster, machi
 	// We can now compare the various AWS state to the state we were passed.
 	// We will check immutable state first, in order to fail quickly before
 	// moving on to state that we can mutate.
-	if a.isMachineOutdated(scope.MachineConfig, instanceDescription) {
-		return errors.Errorf("found attempt to change immutable state")
+	if errs := a.isMachineOutdated(scope.MachineConfig, instanceDescription); len(errs) > 0 {
+		return errors.Errorf("found attempt to change immutable state for machine %q: %v", machine.Name, errs)
```

Comment:
can there be a function to log each validation error separately? this is a bit challenging to read:

https://play.golang.org/p/iq8Yir6efm9

I think this will also get rid of the need to delineate with parentheses.

Reply:
Each error is within `()`. Expect the log to look like `[(Hello world) (second error) (something else)]` @rudoi can confirm

Reply:
you can also do something like this https://play.golang.org/p/we76UelgMXG
```suggestion
		return errors.Errorf("found attempt to change immutable state for machine %q: %+q", machine.Name, errs)
```

Reply:
this will also mean you can get rid of the `()` https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/649/files#r265169842

Reply:
agree with all of the above. updated.

Reply:
there are `%q`'s inside the `%+q` so this will result in some nested quotes in the output, FWIW.

Reply:
I think that should be OK.